tests/test_algorithm.py

```diff
@@ -759,6 +759,48 @@ def test_invalid_order_parameters(self):
         )
         algo.run(self.data_portal)
 
+    def test_cannot_order_in_before_trading_start(self):
+        algotext = """
+from zipline.api import sid
+from zipline.errors import OrderInBeforeTradingStart
```

Comment:
I think semantically (and for easier debugging/tracebacks etc.) the `assertRaises` should be done in the test case method.

This function could be templated like this:

```
 def initialize(context):
     context.asset = sid(133)
 
 def before_trading_start(context, data):
     {order_func}(context.asset, {order_arg_1})
```

Then called with:
```
algot_text = algo_text_template.format(order_func=order_percent,  order_arg_1=1)
algo = TradingAlgorithm(script=algotext, sim_params=self.sim_params,
                                       data_frequency='daily', env=self.env)
with self.assertRaises(OrderInBeforeTradingStart):
    algo.run(self.data_portal)
```